security/pkg/pki/ca/selfsignedcarootcertrotator.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ca
+
+import (
+	"bytes"
+	"encoding/base64"
+	"time"
+
+	"istio.io/istio/security/pkg/k8s/configmap"
+	"istio.io/istio/security/pkg/k8s/controller"
+	"istio.io/istio/security/pkg/monitoring"
+	"istio.io/istio/security/pkg/pki/util"
+	certutil "istio.io/istio/security/pkg/util"
+	"istio.io/pkg/log"
+
+	v1 "k8s.io/api/core/v1"
```

Comment:
Can we make this not dependent on the k8s interface?
Remember our goal is to make Citadel run in native env. Ideally the files other than security/k8s/... should be k8s agnostic. Let's not depend on the k8s interface for the newly added files.

Reply:
OK for now. We can have another PR to fix this.